### PR TITLE
Remove vsts artifacts

### PR DESCRIPTION
**Pull request by jim-minter**
* stop using vsts build artifacts and use docker images instead
* use the ARO-RP git commit instead of having to supply a vsoDeployerBuildID
* use the config git commit instead of having to supply a vsoConfigBuildID
* remove a whole pile of billing-related duplication

### CODE REVIEW

**Comment by jim-minter** [comment]
Untested.  @mjudeikis @Makdaam I wonder whether something like this will help us.  @hawkowl your view here would also be appreciated.

**Comment by jim-minter** [comment]
@hawkowl @mjudeikis repushed.  Still untested.  Still would like a second look.

**Comment by mjudeikis** [comment]
@jim-minter I think im fine with progressing with this, but before we start investing time, do you know if artifacts are involved in any way in Component governance scans? If they are not and we relay on cgmanifest - we can progress

**Comment by jim-minter** [comment]
> artifacts are involved in any way in Component governance scans

To my knowledge, they are not.

File: .pipelines/templates/template-job-deploy-azure-env.yml [hunk 1]
```diff
@@ -10,116 +10,54 @@ parameters:
...
-          REGION:$(REGION)
-        authenticationMethod: OAuth Token
-        password: $(System.AccessToken)
-        enableBuildInQueueCondition: false
-        dependentOnSuccessfulBuildCondition: true
-        dependentOnFailedBuildCondition: true
-        checkbuildsoncurrentbranch: false
-        failTaskIfConditionsAreNotFulfilled: true
+          - checkout: |
+              git://AzureRedHatOpenShift/RP-Config@${{ parameters.rpConfigCommit }}
```
Comment (mjudeikis) [inline .pipelines/templates/template-job-deploy-azure-env.yml:27]: This does not work with:
image
 
There is some issues how VSTS handles checkouts with inline syntax and commit hash'es

Comment (mjudeikis) [reply .pipelines/templates/template-job-deploy-azure-env.yml:27]: I have a feeling it does it via ref not commit resolution ;/

**Review by mjudeikis — CHANGES_REQUESTED** [review]
@jim-minter let's discuss this. I think VSTS is a bit limited in what we can do ;/

**Comment by jim-minter** [comment]
Abandoning this for now - revisit as part of ev2 work

**Pull request closed without merging**